src/Hedgehog.Experimental.Tests/GenTests.fs

```diff
@@ -845,6 +909,18 @@ module ShrinkTests =
             rendered.Contains "[[1]\n [0]" ||
             rendered.Contains "[[1]]"@>
 
+  open System.Text.RegularExpressions
+  [<Fact>]
+  let ``auto of recursive ResizeArray shrinks correctly`` () =
+    let property = property {
+      let! rra = GenX.auto<RecResizeArray>
+      test <@ not (rra.X.Count = 2) @>
+    }
+    let shrunkenLine = (render property).Split('\n').[1]
+    let regex = {RecResizeArray.X = ResizeArray()} |> sprintf "%A" |> Regex.Escape |> Regex
+    let matches = regex.Matches shrunkenLine
+    test <@ 2 = matches.Count @>
```

Comment:
I don't understand what this test is checking.  I think it is too complicated.

Instead, I added a simple test in PR https://github.com/dharmaturtle/fsharp-hedgehog-experimental/pull/1 (that is merging code into this branch in your fork).  Then I make that test pass by calling the constructor last.

Reply:
> I think it is too complicated.

Agreed - I'll delete it.

> I made that test pass by calling the constructor last.

Lemme see if I'm understanding why this must be:

* `ICollection.Add` mutates. If it didn't mutate, then we wouldn't need to do this.
* When shrinking, the property computation expression (CE) may reuse the initial parts of the CE. Evidence:
![image](https://user-images.githubusercontent.com/1278803/134811607-70f909df-1e07-4913-ac49-663b923b0684.png)
* In my second version where I constructed the collection _first_, this would lead to the _reuse_ of that collection. In particular, the `for` loop would mutate the same underlying collection in subsequent shrinking runs, leading the shrinking algorithm astray.

Sound about right?

Reply:
Yes